Source/WebCore/css/SelectorChecker.cpp

```diff
@@ -350,8 +361,8 @@ SelectorChecker::MatchResult SelectorChecker::matchRecursively(CheckingContext&
                 return MatchResult::fails(Match::SelectorFailsCompletely);
 
             auto pseudoId = CSSSelector::pseudoId(context.selector->pseudoElement());
-            if (pseudoId != PseudoId::None)
-                dynamicPseudoIdSet.add(pseudoId);
+            if (pseudoId)
+                dynamicPseudoIdSet.add(*pseudoId);
```

Comment:
nit: you could merge the declaration and the if in one line:

`if (auto pseudoId = CSSSelector::pseudoId(context.selector->pseudoElement()))`